README.md

```diff
@@ -1,16 +1,130 @@
-# Title
+## JavaScript Advanced Functions: The Collection-Processing Methods Family Tree
 
 ## Learning Goals
 
--SWBAT 1
--SWBAT 2
+* Use `map` to transform an `Array`
+* Use `reduce` to reduce an `Array` to a value
+* Use JavaScript documentation to learn more about other variations on `map` and `reduce`
+* Use `filter` to filter an `Array`
 
 ## Introduction
 
-## SWBAT 1
+Now that you have written `map` and `reduce`, here's the big reveal: JavaScript
+_already_ has these methods built into its `Array` data type!
 
-## SWBAT 2
+
+## Use `map` to Transform an `Array`
+
+```ruby
+[10, 20, 30, 40].map(function(a){return a * 2}) //=> [20, 40, 60, 80]
+```
+
+## Use `reduce` to Reduce an `Array` to a Value
+
+```ruby
+[10, 20, 30, 40].reduce(function(i, memo){ return i + memo }) //=> 100
+[10, 20, 30, 40].reduce(function(i, memo){ return i + memo }, 100) //=> 200
+```
+
+Remember, ***JavaScript loves functions*** by being able to pass functions
+around comfortably, we can take advantage of methods that JavaScript gives us!
+Given what you know about writing your own `map` and `reduce` functions,
+read the JavaScript documentation and make sure you know how to use the
+versions JavaScript provides you:
+
+* [map][]
+* [reduce][]
+
+## Use JavaScript Documentation to Learn More About Other Variations on `map` and `reduce`
+
+Now that we understand the "Character of Collection-Processing Methods" we are
+ready to unleash the full power of this module!  Instead of merely repeating
+the documentation, we're going to help you apply your understanding to the
+`filter` method, and then we're going to give you a list of document resources
+you should look up and master.
+
+## Filter
+
+Look at the documentation for [filter][].  The syntax snippet is provided as:
+
+```js
+var newArray = arr.filter(callback(element[, index[, array]])[, thisArg])
+```
+
+Here, we're told that on an `Array`, (`arr`) we add a `.filter` and we provide
+a `callback` and a `thisArg`. Remember, because of JavaScript's non-enforcement
+of _arity_, we don't **have to** provide all the arguments.  We'll talk a lot
+more about `this` and `thisArg` shortly, but for the moment, we can _leave it
+out_.
+
+Let's look some more about the `callback`. The documentation provides what
+parameters are supported by the callback. JavaScript will hand off, per each
+element, to the callback an `element`, the `index` number of that element, and
+`array`, the whole darn `Array` itself. Yet again, because of _arity_
```

Comment:
is this an emphasis like _Yes_ the whole darn thing! I think separate to two sentences to avoid confusion that there are two arrays going into the callback